core/drivers/crypto/crypto_api/include/drvcrypt.h

```diff
@@ -0,0 +1,69 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   Crypto Driver exported constants and interfaces.
+ */
+#ifndef __DRVCRYPT_H__
+#define __DRVCRYPT_H__
+
+/* Global includes */
+#include <tee_api_types.h>
+
+#ifdef CFG_CRYPTO_DRV_DBG
+#define CRYPTO_TRACE	DMSG
+#else
+#define CRYPTO_TRACE(...)
+#endif
+
+/*
+ * Crypto Library Algorithm enumeration
+ */
+enum drvcrypt_algo_id {
+	CRYPTO_HASH = 0,     /* HASH driver */
+	CRYPTO_HMAC,         /* HMAC driver */
+	CRYPTO_MAX_ALGO      /* Maximum number of algo supported */
+};
+
+/*
+ * Cryptographic buffer type
+ */
+struct drvcrypt_buf {
+	uint8_t *data;   /* Pointer to the data buffer */
+	size_t  length;  /* Length in bytes of the data buffer */
+};
+
+/*
+ * Cryptographic module registration
+ *
+ * Inputs:
```

Comment:
prefer without `Inputs:` and with `@` to highlight argument labels and some rephrasing (s/idx/algo_id/).
```
/*
 * Cryptographic module registration
 *
 * @id  Identifier for the cryptographic algorithm family 
 * @ops  Registered cryptographic module operation handlers
 ```

Reply:
ok I'll review everywhere